src/connection-manager/connections.ts

```diff
@@ -64,16 +64,15 @@ async function setHandlers(ip: string, ws: WebSocket): Promise<void> {
     ws.on('close', () => {
       if (connections.get(ip)?.url === ws.url) {
         connections.delete(ip)
+        void saveNodeWsUrl(ws.url, false)
       }
-      void saveNodeWsUrl(ws.url, false)
       ws.terminate()
       resolve()
     })
     ws.on('error', () => {
       if (connections.get(ip)?.url === ws.url) {
         connections.delete(ip)
       }
-      void saveNodeWsUrl(ws.url, false)
       ws.terminate()
       resolve()
```

Comment:
Will an `error` cause the `close` callback handler to be called? Because if not, we should also mark the connection as disconnected on Error

Reply:
Yes the error causes the close callback handler to be called.